mlflow/pyfunc/__init__.py

```diff
@@ -946,15 +966,52 @@ def load_model(
 
     _add_code_from_conf_to_system_path(local_path, conf, code_key=CODE)
     data_path = os.path.join(local_path, conf[DATA]) if (DATA in conf) else local_path
-    model_config = _get_overridden_pyfunc_model_config(
-        conf.get(MODEL_CONFIG, None), model_config, _logger
-    )
+
+    if isinstance(model_config, str):
+        model_config = _validate_and_get_model_config_from_file(model_config)
+
+    conf_model_config = conf.get(MODEL_CONFIG, None)
+    if isinstance(conf_model_config, str):
+        conf_model_config_path = os.path.join(local_path, os.path.basename(conf_model_config))
+        conf_model_config = _validate_and_get_model_config_from_file(conf_model_config_path)
+
+    model_config = _get_overridden_pyfunc_model_config(conf_model_config, model_config, _logger)
+    if model_config:
+        conf.update({MODEL_CONFIG: model_config})
 
     try:
-        if model_config:
-            model_impl = importlib.import_module(conf[MAIN])._load_pyfunc(data_path, model_config)
-        else:
+        if "langchain" in conf[MAIN]:
             model_impl = importlib.import_module(conf[MAIN])._load_pyfunc(data_path)
+        elif "pyfunc" in conf[MAIN] and MODEL_CODE_PATH in conf:
+            flavor_code_path = conf.get(MODEL_CODE_PATH)
+            code_path = os.path.join(local_path, os.path.basename(flavor_code_path))
+
+            model = _load_model_code_path(code_path, model_config)
+
+            artifacts = {}
+            for saved_artifact_name, saved_artifact_info in conf.get(
+                CONFIG_KEY_ARTIFACTS, {}
+            ).items():
+                artifacts[saved_artifact_name] = os.path.join(
+                    data_path, saved_artifact_info[CONFIG_KEY_ARTIFACT_RELATIVE_PATH]
+                )
+
+            context = PythonModelContext(artifacts=artifacts, model_config=model_config)
+            model.load_context(context=context)
+            signature = mlflow.models.Model.load(data_path).signature
+
+            model_impl = _PythonModelPyfuncWrapper(
+                python_model=model,
+                context=context,
+                signature=signature,
+            )
```

Comment:
Instead of moving all of this logic directly into `mlflow.pyfunc.load_model()`, can we define  a new loader module for code-based custom pyfunc models similar to the loader module for class-based custom pyfunc models (see https://github.com/mlflow/mlflow/blob/f75d65de6efb912fdd77fb8e491276853f154c81/mlflow/pyfunc/model.py#L483)? That way, we can keep the existing logic the same and just call `_load_pyfunc()`.